src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1472,7 +1472,7 @@ public static IEnumerable<int> Range(int start, int count)
 
         private static IEnumerable<int> RangeIterator(int start, int count)
         {
-            for (int i = 0; i < count; i++) yield return start + i;
+            for (int end = start + count; start != end; start++) yield return start;
```

Comment:
Is it ever possible for count to come in here as negative? If so this loop will do bad things. Why do we think this is more so much more efficient?